src/app/ReadClient.cpp

```diff
@@ -0,0 +1,249 @@
+    VerifyOrExit(mpExchangeCtx != nullptr, err = CHIP_ERROR_NO_MEMORY);
+    mpExchangeCtx->SetResponseTimeout(kImMesssageTimeout);
+
+    err = mpExchangeCtx->SendMessage(Protocols::InteractionModel::MsgType::ReadRequest, std::move(msgBuf),
+                                     Messaging::SendFlags(Messaging::SendMessageFlags::kExpectResponse));
+    SuccessOrExit(err);
+    MoveToState(ClientState::ReadRequestSending);
+
+exit:
+    ChipLogFunctError(err);
+    return err;
+}
+
+void ReadClient::OnMessageReceived(Messaging::ExchangeContext * apExchangeContext, const PacketHeader & aPacketHeader,
+                                   const PayloadHeader & aPayloadHeader, System::PacketBufferHandle aPayload)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    VerifyOrDie(apExchangeContext == mpExchangeCtx);
+
+    ClearExistingExchangeContext();
+
+    err = ProcessReportData(std::move(aPayload));
+    SuccessOrExit(err);
+
+exit:
+    Reset();
+}
+
+CHIP_ERROR ReadClient::ClearExistingExchangeContext()
+{
+    if (mpExchangeCtx != nullptr)
+    {
+        mpExchangeCtx->Abort();
+        mpExchangeCtx = nullptr;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ReadClient::ProcessReportData(System::PacketBufferHandle aPayload)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    ReportData::Parser report;
+
+    bool isEventListPresent  = false;
+    bool suppressResponse    = false;
+    bool moreChunkedMessages = false;
+
+    chip::System::PacketBufferTLVReader reader;
+
+    reader.Init(std::move(aPayload));
+    reader.Next();
+
+    err = report.Init(reader);
+    SuccessOrExit(err);
+
+    err = report.CheckSchemaValidity();
+    SuccessOrExit(err);
+
+    err = report.GetSuppressResponse(&suppressResponse);
+    if (CHIP_END_OF_TLV == err)
+        err = CHIP_NO_ERROR;
+    SuccessOrExit(err);
+
+    err = report.GetMoreChunkedMessages(&moreChunkedMessages);
+    if (CHIP_END_OF_TLV == err)
+        err = CHIP_NO_ERROR;
+    SuccessOrExit(err);
+
+    {
+        EventList::Parser eventList;
+
+        err = report.GetEventDataList(&eventList);
+        if (CHIP_NO_ERROR == err)
+        {
+            isEventListPresent = true;
+        }
+        else if (CHIP_END_OF_TLV == err)
+        {
+            isEventListPresent = false;
+            err                = CHIP_NO_ERROR;
+        }
+        SuccessOrExit(err);
+
+        if (isEventListPresent && nullptr != mpDelegate && !moreChunkedMessages)
+        {
+            // re-initialize the reader (reuse to save stack depth).
+            eventList.GetReader(&reader);
+            err = mpDelegate->HandleEventStreamReceivedCallBack(mpExchangeCtx, &reader);
+            SuccessOrExit(err);
+        }
+    }
+
+    if (!suppressResponse)
+    {
+        // TODO: Add status report support and correspond handler in ReadHandler, particular for situation when there
+        // are multiple reports
+    }
+    if (nullptr != mpDelegate && !moreChunkedMessages)
```

Comment:
Is there something tracking actually making the `moreChunkedMessages` case work?  Please make sure there's an issue filed.

Reply:
just file new ticket, thanks